src/ui/public/kuery/lib/ast.js

```diff
@@ -0,0 +1,39 @@
+import grammar from 'raw!./kuery.peg';
+import PEG from 'pegjs';
+import _ from 'lodash';
+import { nodeTypes } from './node_types';
+
+const kueryParser = PEG.buildParser(grammar);
+
+export function fromKueryExpression(expression) {
```

Comment:
It would be great to have unit tests ensuring that
```javascript
someKueryExpression === toKueryExpression(fromKueryExpression(someKueryExpression))
```
for a "canonically formatted" kuery of each node type.